fastlane/lanes/release.rb

```diff
@@ -0,0 +1,225 @@
+# frozen_string_literal: true
+
+# Lanes related to the Release Process (Code Freeze, Betas, Final Build, AppStore Submission…)
+#
+platform :ios do
+
+  # Executes the initial steps needed during code freeze
+  # - Cuts a new release branch
+  # - Extract the Release Notes
+  # - Freezes the GitHub milestone and enable GitHub branch protection for the new branch
+  #
+  # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
+  #
+  desc 'Executes the initial steps needed during code freeze'
+  lane :code_freeze do |options|
+    gutenberg_dep_check
+    ios_codefreeze_prechecks(options)
+
+    ios_bump_version_release(skip_deliver: true)
+    new_version = ios_get_app_version
+
+    release_notes_source_path = File.join(PROJECT_ROOT_FOLDER, 'RELEASE-NOTES.txt')
+    extract_release_notes_for_version(
+      version: new_version,
+      release_notes_file_path: release_notes_source_path,
+      extracted_notes_file_path: extracted_release_notes_file_path(app: :wordpress)
+    )
+    # It would be good to update the action so that it can:
+    #
+    # - Use a custom commit message, so that we can differentiate between
+    #   WordPress and Jetpack
+    # - Have some sort of interactive mode, where the file is extracted and
+    #   shown to the user and they can either confirm and let the lane commit,
+    #   or modify it manually first and then run through the
+    #   show-confirm-commit cycle again
+    #
+    # In the meantime, we can make due with a duplicated commit message and the
+    # `print_release_notes_reminder` at the end of the lane to remember to make
+    # updates to the files.
+    extract_release_notes_for_version(
+      version: new_version,
+      release_notes_file_path: release_notes_source_path,
+      extracted_notes_file_path: extracted_release_notes_file_path(app: :jetpack)
+    )
+    ios_update_release_notes(new_version: new_version)
+
+    setbranchprotection(repository: GHHELPER_REPO, branch: "release/#{new_version}")
+    setfrozentag(repository: GHHELPER_REPO, milestone: new_version)
+    ios_check_beta_deps(podfile: File.join(PROJECT_ROOT_FOLDER, 'Podfile'))
+
+    print_release_notes_reminder
+  end
+
+  # Completes the final steps for the code freeze
+  #  - Generate `.strings` files from code then merge the other, manually-maintained `.strings` files with it
+  #  - Trigger the build of the first beta on CI
+  #
+  # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
+  #
+  desc 'Completes the final steps for the code freeze'
+  lane :complete_code_freeze do |options|
```

Comment:
@mokagio Can I get a preliminary review (even if PR is still draft) and feedback on the direction I'm going towards here, especially w.r.t the new, less verbose and less visually-noisy lanes' documentation comments

---

Other ideas I've tried include:

1️⃣  Moving all the lane docs into the `desc` string (after all, it's supposed to be what `desc` is for), using things like this:

```ruby
  desc <<~DESC
    Completes the final steps for the code freeze
     - Generate `.strings` files from code then merge the other, manually-maintained `.strings` files with it
     - Trigger the build of the first beta on CI
     
      @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
  DESC
  lane :complete_code_freeze do |options|
```

Which has the benefit of showing the whole help in the console when you run `fastlane lanes` to list them all… but has the big feedback imho that the documentation is then seen as a regular `String` by your IDE's syntax highlighter (and not a comment anymore), and thus will not pop up as nicely nor be as easily differentiated by color as when it is written as a `# comment`… so I ended up preferring keeping it as a `# comment` after all.

2️⃣  Also tried a mix to avoid repeating the `desc` tagline, like this:

```ruby
  desc 'Completes the final steps for the code freeze'
  #  - Generate `.strings` files from code then merge the other, manually-maintained `.strings` files with it
  #  - Trigger the build of the first beta on CI
  #
  # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
  #
  lane :complete_code_freeze do |options|
```

But again, that looks odd in the IDE (like VSCode) as it ends up appearing as a mix of colors (the color for string literals for the `desc` line — e.g. red in my IDE — but then the color for comments for the doc details — green in my IDE — and that does not look super nicer either when trying to skim the whole file to immediately spot new lanes boundaries…

Which is why I was leaning towards a format like what's in the current diff instead, less noisy that all those `####…` we have been having, but still with a nice way of immediately spotting the lane documentation by it all being in the comments (e.g. green) color in its entirety.

_tbh I'm not super happy with the `desc` often being a repetition of the first line of the doc-comment (though it's not always the case, note that sometimes I've used a `desc` which is more concise), but overall I think it's still the thing that gives the nicer rendering in most IDEs to make it a delight to skim and navigate the file day to day._